### PR TITLE
improve robustness of eventviewer

### PR DESCRIPTION
**Pull request by yrenier**
Currently the event viewer crashes if one is missing the trigger output for example in the container.
This small modifications should prevent such crashes by just ignoring the offending plot request while printing some warning to the console.

### CODE REVIEW

**Comment by yrenier** [comment]
I forgot to put this into context: this PR allows to use the viewer with MC events using @jurysek 's branch jurysek_simtel .

**Comment by dneise** [comment]
Looks good to me on first glance .. @jurysek did you try this out once? Does it work for you? Then we can maybe merge it.

**Comment by dneise** [comment]
Just one comment. 
digicampipe nowadays installs a command line tool called `digicam-view` when it is installed.
the imeplementation is here:
https://github.com/cta-sst-1m/digicampipe/blob/master/digicampipe/scripts/digicamview.py

And the `EventViewer` which is used in that tool does not come from this repo anymore, instead it comes from `digicampipe/visualization`.
https://github.com/cta-sst-1m/digicampipe/blob/master/digicampipe/visualization/__init__.py

So .. maybe we want to modify that `EventViewer` instead of this one here. 

This was done in reaction to #5

**Comment by dneise** [comment]
Sorry for not reacting quicker on this PR .. I somehow missed it entirely.

File: digicamviewer/viewer.py [hunk 1]
```diff
@@ -126,37 +154,46 @@ def next(self, event=None, step=1):
...
+            else:
+                self.flag = np.nan
+            if self.r1_container.nsb is not None:
+                self.nsb = self.r1_container.nsb
+            else:
+                self.nsb = np.nan * zero_image
+            if self.r1_container.gain_drop is not None:
+                self.gain_drop = self.r1_container.gain_drop
+            else:
+                self.gain_drop = np.nan * zero_image
```
Comment (dneise) [inline digicamviewer/viewer.py:182]: thanks for rewriting these ifs .. much better readable now.

File: digicamviewer/viewer.py [hunk 2]
```diff
@@ -167,254 +204,205 @@ def draw(self):
...
             self.axis_readout.set_ylabel('[LSB]')
 
     def compute_trace(self):
-
+        image = None
         if self.readout_view_type in self.readout_view_types:
-
             if self.readout_view_type == 'raw':
-
+                print('trace:', self.adc_samples)
```
Comment (dneise) [inline digicamviewer/viewer.py:273]: maybe this debug print should be deleted some time

**Comment by yrenier** [comment]
I will take a look at digicam-view and push there then.

**Comment by jurysek** [comment]
I have test it and it works nice for me!

**Comment by yrenier** [comment]
I pushed the changes in digicam (branch viewer_fail_nicely).
@jurysek could you test it there as well ?
Also, should we get rid of digicamviewer all together if is is already included in digicampipe ?

**Comment by jurysek** [comment]
Checked, the viewer that You have pushed into digicampipe works for me as well.

**Comment by yrenier** [comment]
Thanks for checking !

**Pull request closed without merging**